### ISSUE TITLE
Prerlease version comparisons bug: old advisories incorrectly applied to new versions

### ISSUE THREAD

**Issue by tl-alex-butler** [rustsec/rustsec#300]
Running `cargo audit` with recently published _actix-web_ `4.0.0-beta.1` is producing errors due to older advisories. Presumably this is a version comparison bug?

Update: Caused by: https://github.com/RustSec/rustsec-crate/issues/218

## Reproduce
```toml
# cargo.toml
[dependencies]
actix-web = "4.0.0-beta.1"
```

Run `cargo audit`

```
cargo audit
    Fetching advisory database from `https://github.com/RustSec/advisory-db.git`
      Loaded 175 security advisories (from /home/alex/.cargo/advisory-db)
    Updating crates.io index
    Scanning Cargo.lock for vulnerabilities (166 crate dependencies)
Crate:         actix-codec
Version:       0.4.0-beta.1
Title:         Use-after-free in Framed due to lack of pinning
Date:          2020-01-30
ID:            RUSTSEC-2020-0049
URL:           https://rustsec.org/advisories/RUSTSEC-2020-0049
Solution:      Upgrade to >=0.3.0-beta.1
Dependency tree: 
actix-codec 0.4.0-beta.1
├── awc 3.0.0-beta.1
│   └── actix-web 4.0.0-beta.1
│       └── rusttmp 0.1.0
├── actix-web 4.0.0-beta.1
├── actix-utils 3.0.0-beta.1
│   ├── actix-web 4.0.0-beta.1
│   ├── actix-tls 3.0.0-beta.2
│   │   └── actix-http 3.0.0-beta.1
│   │       ├── awc 3.0.0-beta.1
│   │       └── actix-web 4.0.0-beta.1
│   ├── actix-server 2.0.0-beta.2
│   │   └── actix-web 4.0.0-beta.1
│   └── actix-http 3.0.0-beta.1
├── actix-tls 3.0.0-beta.2
├── actix-server 2.0.0-beta.2
└── actix-http 3.0.0-beta.1

Crate:         actix-http
Version:       3.0.0-beta.1
Title:         Use-after-free in BodyStream due to lack of pinning
Date:          2020-01-24
ID:            RUSTSEC-2020-0048
URL:           https://rustsec.org/advisories/RUSTSEC-2020-0048
Solution:      Upgrade to >=2.0.0-alpha.1
Dependency tree: 
actix-http 3.0.0-beta.1
├── awc 3.0.0-beta.1
│   └── actix-web 4.0.0-beta.1
│       └── rusttmp 0.1.0
└── actix-web 4.0.0-beta.1

Crate:         actix-web
Version:       4.0.0-beta.1
Title:         Multiple memory safety issues
Date:          2018-06-08
ID:            RUSTSEC-2018-0019
URL:           https://rustsec.org/advisories/RUSTSEC-2018-0019
Solution:      Upgrade to >=0.7.15
Dependency tree: 
actix-web 4.0.0-beta.1
└── rusttmp 0.1.0

...
```

**Comment by tarcieri** [comment]
Note: this is tracked as https://github.com/RustSec/rustsec-crate/issues/218

**Comment by forbidden404** [comment]
I'm still getting this error even after https://github.com/RustSec/rustsec/pull/389 got merged, is there any other work necessary for this issue to be closed? `cargo-audit` is currently on `v0.14.1` on my machine.

**Comment by tarcieri** [comment]
Should be addressed in the v0.15.0 release: https://github.com/RustSec/rustsec/pull/392
